simphony_mayavi/add_dataset.py

```diff
@@ -0,0 +1,65 @@
+from mayavi import mlab
+
+from simphony.cuds.abc_mesh import ABCMesh
+from simphony.cuds.abc_particles import ABCParticles
+from simphony.cuds.abc_lattice import ABCLattice
+
+from simphony_mayavi.sources.api import CUDSSource
+
+
+def add_dataset(cuds, point_scalars=None, point_vectors=None,
+                cell_scalars=None, cell_vectors=None):
+    ''' Add CUDS dataset to the current scene.
+    If no scene is defined, a new scene is created.
+
+    Parameters
+    ----------
+    cuds : ABCMesh, ABCParticles, ABCLattice
+
+    point_scalars : str, optional
+        CUBA name of the data to be selected as point scalars.
+        Default is the first available point scalars.
+
+    point_vectors : str, optional
+        CUBA name of the data to be selected as point vectors.
+        Default is the first available point vectors.
+
+    cell_scalars : str, optional
+        CUBA name of the data to be selected as cell scalars.
+        Default is the first available cell scalars.
+
+    cell_vectors : str, optional
+        CUBA name of the data to be selected as cell vectors.
+        Default is the first available cell vectors.
+
+    Returns
+    -------
+    source : CUDSSource
+
+    Notes
+    -----
+    To turn off visualisation of a point/cell data, assign
+    [point/cell]_[scalars/vectors] to an empty string
+    '''
+    if isinstance(cuds, (ABCMesh, ABCParticles, ABCLattice)):
```

Comment:
The type checking in add_dataset is indeed redundant here because Traits already does it.  I will remove it.  For `show` and `snapshot`, I will try to eliminate this by catching the TraitError and thereby preventing a figure from opening if the Source cannot be created.